CMakeLists.txt

```diff
@@ -347,7 +349,12 @@ if(Python3_FOUND AND Git_FOUND)
       RESULT_VARIABLE SUBMODULE_INIT_RESULT
     )
     if(NOT SUBMODULE_INIT_RESULT EQUAL "0")
-      message(FATAL_ERROR "check_submodule_init.py failed, see the logs above")
+      set(_CHECK_SUBMODULE_MSG "check_submodule_init.py failed, see the logs above")
+      if(IREE_ERROR_ON_MISSING_SUBMODULES)
```

Comment:
I'd also be okay with this check moving up to the top level (so `IREE_ERROR_ON_MISSING_SUBMODULES AND Python3_FOUND AND Git_FOUND`). The check can be slow (~5 seconds), and anyone setting this option should be well aware of their submodule state anyways.

Reply:
Yeah if someone is setting this option I think they don't need the warning spam

Reply:
If you're both okay with it, I am happy to move it up :) Done.